### PR TITLE
Feat/refer page

### PR DESCRIPTION
**Pull request by Josh-Alhassan**
edustipend-refer-page

### CODE REVIEW

**Review by edustipend-root — CHANGES_REQUESTED** [review]
Hello @Josh-Alhassan, thanks for working on this and raising a PR quickly. The changes look good for the most part but I've added some comments on some minor changes.

Also, I didn't see you implement the copy functionality. Was that to be in a separate follow-up PR?

Finally, there are some dissimilarities between your UI and the Figma design. Please take a look at the screenshots below (pay attention to the input fields):

**Your UI**
image

**Figma Design**
image

Please make the changes and let me know once you've pushed them so I can review and merge this in.

File: src/pages/refer-page/ReferPageText.js
[the remark this answers is not in the document]
Comment (Josh-Alhassan) [reply src/pages/refer-page/ReferPageText.js:1]: image

I just checked mine and it's the same as the Figam Design. I think it's a rendering issue, maybe at your end. But I will look into it with other devices to make sure it's all consistent.

File: src/pages/refer-page/ReferPageText.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const referPageTexts = {
```
[the remark this answers is not in the document]
Comment (Josh-Alhassan) [reply src/pages/refer-page/ReferPageText.js:1]: referPageTexts, renamed to referralPageCopy

File: src/pages/refer-page/ReferPageText.js
[the remark this answers is not in the document]
Comment (Josh-Alhassan) [reply src/pages/refer-page/ReferPageText.js:1]: `ReferPageText.js` is changed to `constants.js`

File: src/routesConstant.js [hunk 2]
```diff
@@ -14,5 +14,6 @@ export const routesConstant = {
   DONATION: '/donation',
   REPORTS: '/impacts/reports',
   IMPACTS: '/impacts',
-  TRANSPARENCY_DASHBOARD: '/support-a-learner/transparency-dashboard'
+  TRANSPARENCY_DASHBOARD: '/support-a-learner/transparency-dashboard',
+  REFERRAL_PAGE: '/referral-page'
```
[the remark this answers is not in the document]
Comment (Josh-Alhassan) [reply src/routesConstant.js:18]: Route path update to `/support-a-learner/refer-a-friend`

File: src/pages/refer-page/ReferPage.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+describe('ReferPage', () => {
+  test('renders ReferPage correctly', () => {
+    render(<ReferPage />);
+
+    expect(screen.getByText(referPageTexts.referHeader)).toBeInTheDocument();
+
+    expect(screen.getByText(referPageTexts.referParagraph)).toBeInTheDocument();
+
+    expect(screen.getByText('Generate Link')).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (Josh-Alhassan) [reply src/pages/refer-page/ReferPage.test.js:14]: Refactored. "Generate Link" text no longer hardcoded

File: src/components/FormInput/FormInput.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import styles from './FormInput.module.css';
+import PropTypes from 'prop-types';
+
+export default function FormInput({ label, placeholder, type }) {
+  return (
+    <div className={styles.formField}>
+      <label className={styles.formLabel} htmlFor="name">
+        {label}
+      </label>
+      <input className={styles.inputField} type={type} placeholder={placeholder} />
```
[the remark this answers is not in the document]
Comment (Josh-Alhassan) [reply src/components/FormInput/FormInput.jsx:10]: I have deleted the input component I created and imported the already created `<Input />`  component. It's fixed now.

File: src/pages/refer-page/ReferPageText.js
[the remark this answers is not in the document]
Comment (Josh-Alhassan) [reply src/pages/refer-page/ReferPageText.js:1]: @edustipend-root The "copy" functionality is done. I think all corrections have been fixed.

File: src/pages/refer-page/ReferPage.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        </div>
+
+        <div className={styles.referFormSection}>
+          <div className={styles.referHeader}>
+            <Header size="medium">{referralPageCopy.referHeader}</Header>
+          </div>
+          <p className={styles.referParagraph}>{referralPageCopy.referParagraph}</p>
+
+          <form action="" className={styles.referForm}>
+            <Input value="Enter name" label="Name" placeholder="Enter name" className={styles.entry} />
```
Comment (edustipend-root) [inline src/pages/refer-page/ReferPage.jsx]: @Josh-Alhassan, this is not correct - you are setting the value of the input field to `"Enter name"`. How is this meant to update when the user types in a new value? 

Same applies for the email field below.

Comment (edustipend-root) [reply src/pages/refer-page/ReferPage.jsx]: Did you actually test this out that it works?

Comment (edustipend-root) [reply src/pages/refer-page/ReferPage.jsx]: @Josh-Alhassan, I pushed a commit to fix it - so we should be good now.

**Review by edustipend-root — APPROVED** [review]
The work to make the API call to get the link will be in a follow-up PR once the BE is implemented.

**Pull request merged**